tests/test_maxcut.py

```diff
@@ -80,6 +81,20 @@ def test_maxcut_weighted_qaoa_obj():
         assert np.isclose(precomputed_cuts.dot(np.abs(sv) ** 2), row["Expected cut of QAOA"])
 
 
+def test_maxcut_precompute():
+    N = 4
+    G = nx.random_regular_graph(3, N)
+    print(G.edges())
+    for u, v, w in G.edges(data=True):
+        w["weight"] = np.random.rand()
+    precomputed_cuts = precompute_energies(maxcut_obj, N, w=get_adjacency_matrix(G))
+    simclass = qokit.fur.choose_simulator("gpu")
```

Comment:
Shouldn't this function call fail if no GPU is present? I checked on a CPU-only instance, and it does not throw an error

Reply:
`precompute_energies` is the CPU-only function, so it shouldn't fail. If you use the simulator from `choose_simulator("gpu")` it should fail on initialization.

Reply:
Github comments are a bit ambiguous; I was referring to line 91 in particular. Running the following two lines in a CPU-only instance does not throw an error:
```
import qokit
qokit.fur.choose_simulator("gpu")
```

Reply:
Yes, this is expected, since `choose_simulator` returns the class, which is then used in later steps of simulation. Using this class will raise an error